geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/ParameterRequirements/ZAddParameterRequirements.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.redis.internal.ParameterRequirements;
+
+import static org.apache.geode.redis.internal.RedisConstants.ERROR_INVALID_ZADD_OPTION_NX_XX;
+import static org.apache.geode.redis.internal.RedisConstants.ERROR_NOT_A_VALID_FLOAT;
+import static org.apache.geode.redis.internal.RedisConstants.ERROR_SYNTAX;
+
+import java.util.Iterator;
+import java.util.List;
+
+import org.apache.geode.redis.internal.netty.Coder;
+import org.apache.geode.redis.internal.netty.Command;
+import org.apache.geode.redis.internal.netty.ExecutionHandlerContext;
+
+public class ZAddParameterRequirements implements ParameterRequirements {
+  @Override
+  public void checkParameters(Command command, ExecutionHandlerContext context) {
```

Comment:
The contents of this method seem very similar to what's being done in `ZAddExecutor.executeCommand()`. Would it be possible to avoid iterating everything twice and having duplicate code by using a different approach, or by pulling some of the code out into shared methods?

Reply:
The most obvious way would be to get rid of the ZAddParameterRequirements class, and just verify minimum param count. Then do all the validation in the executor. That is the simplest thing that'll fit into our current execution scheme, but at the cost of complicating the executor. It's true we're iterating through things twice, but we don't expect parameter lists to be all that long in the general case. So there may be a performance/readability tradeoff.

If we want to go that route, I'm game, but we might want to discuss it with the team.

Reply:
Yeah, I think a discussion with the team would be a good idea. I don't want to be requesting changes for stuff that's already been settled on at a higher level in terms of design and approach just because I wasn't around when those decisions were made.

Reply:
Per our team discussion, I've removed ZAddParameterRequirements and moved parameter validation and parsing into ZAddExecutor. I've tried to work with the existing refactoring, I *think* it's pretty clear. Comments obviously welcome!